### PR TITLE
Frontend/fix/#488 auto submission of otp for login

### PR DESCRIPTION
**Pull request by ashkriti**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Auto-submission of the OTP for both driver and customer while logging in: when the user enters the 4th digit of the OTP.

## Motivation and Context
https://github.com/nammayatri/nammayatri/issues/488

## How did you test it?
Tested manually

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

Attachment:

https://user-images.githubusercontent.com/61149512/231963839-23ca2370-44e1-40a3-aaac-822b887e4dd0.mov

https://user-images.githubusercontent.com/61149512/231963631-0ab998e7-6ac5-4042-9a1a-0963d4b53d1b.mov

### CODE REVIEW

File: Frontend/ui-customer/src/Screens/OnBoardingFlow/EnterMobileNumberScreen/Controller.purs [hunk 1]
```diff
@@ -119,21 +119,24 @@ eval (MobileNumberEditTextAction (PrimaryEditTextController.TextChanged id value
...
-            pure $ hideKeyboardOnNavigation true 
-            else pure unit
     let newState = state { props = state.props { btnActiveOTP = if length value == 4 then true else false, letterSpacing = if value == "" then 1.0 else 6.0, wrongOTP = if state.props.wrongOTP && value == "" then true else false}
                   , data = state.data { otp = if length value <= 4 then value else state.data.otp }}
+    _ <- if length value == 4 then do
+            pure $ hideKeyboardOnNavigation true
+            updateAndExit newState $ GoToAccountSetUp newState 
+            else continue newState
     _ <- pure $ spy "entermobile number " state.props.letterSpacing
     continue newState
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Screens/OnBoardingFlow/EnterMobileNumberScreen/Controller.purs:141]: @ashkriti can you remove this unreachable case

File: Frontend/ui-customer/src/Screens/OnBoardingFlow/EnterMobileNumberScreen/Controller.purs [hunk 2]
```diff
@@ -119,21 +119,24 @@ eval (MobileNumberEditTextAction (PrimaryEditTextController.TextChanged id value
...
+            else pure $ AfterRender   
+    ]
 
 eval (OTPEditTextAction (PrimaryEditTextController.TextChanged id value)) state = do 
-    _ <- if length value == 4 then do
-            pure $ hideKeyboardOnNavigation true 
-            else pure unit
     let newState = state { props = state.props { btnActiveOTP = if length value == 4 then true else false, letterSpacing = if value == "" then 1.0 else 6.0, wrongOTP = if state.props.wrongOTP && value == "" then true else false}
                   , data = state.data { otp = if length value <= 4 then value else state.data.otp }}
+    _ <- if length value == 4 then do
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Screens/OnBoardingFlow/EnterMobileNumberScreen/Controller.purs:136]: no need for `_ <-`

File: Frontend/ui-driver/src/Screens/EnterMobileNumberScreen/Controller.purs [hunk 3]
```diff
@@ -77,7 +77,12 @@ eval (PrimaryEditTextAction (PrimaryEditText.TextChanged valId newVal)) state =
...
                                     Nothing -> true 
-  continue state { props = state.props { btnActive = if (length newVal == 10 && isValidMobileNumber) then true else false
+  let newState = state { props = state.props { btnActive = if (length newVal == 10 && isValidMobileNumber) then true else false
                                         , isValid = not isValidMobileNumber }
                                         , data = state.data { mobileNumber = if length newVal <= 10 then newVal else state.data.mobileNumber}}
+  continueWithCmd newState[
+    do
+        if(length newVal == 10 && isValidMobileNumber) then pure $ PrimaryButtonActionController (PrimaryButton.OnClick)
+          else pure $ AfterRender
+  ]
```
Comment (ashishsingh101) [inline Frontend/ui-driver/src/Screens/EnterMobileNumberScreen/Controller.purs:87]: no need to call `continueWithCmd`, you can directly call `exit (GoToNextScreen newState)` or `continue state` as per requirement

Comment (ashishsingh101) [reply Frontend/ui-driver/src/Screens/EnterMobileNumberScreen/Controller.purs:87]: same in other places also

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Screens/OnBoardingFlow/EnterMobileNumberScreen/Controller.purs [hunk 4]
```diff
@@ -119,23 +119,22 @@ eval (MobileNumberEditTextAction (PrimaryEditTextController.TextChanged id value
                                                     if value=="5000500050" then true else false 
                                                         else true 
                                     Nothing -> true 
+    let newState = state { props = state.props { isValidMobileNumber = isValidMobileNumber
+                                            , btnActiveMobileNuber = if (length value == 10 && isValidMobileNumber) then true else false}
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Screens/OnBoardingFlow/EnterMobileNumberScreen/Controller.purs:123]: @ashkriti can you correct the spelling of this variable "btnActiveMobileNuber" => "btnActiveMobileNumber"

**Review by MercyQueen — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Screens/OnBoardingFlow/EnterMobileNumberScreen/Controller.purs (hunk 4 above)
Comment (ashkriti) [reply Frontend/ui-customer/src/Screens/OnBoardingFlow/EnterMobileNumberScreen/Controller.purs]: Done

**Review by ashishsingh101 — APPROVED** [review]

**Comment by Vignesh-772** [comment]
not able see any changes in EnterMobileNumberScreen in Driver

**Comment by ashkriti** [comment]
> not able see any changes in EnterMobileNumberScreen in Driver

In both driver and customer now auto submission of mobile number is not required only auto submission of OTP is required

**Review by Vignesh-772 — APPROVED** [review]

**Review by MercyQueen — APPROVED** [review]

**Pull request merged**